modules/Bio/EnsEMBL/Variation/DBSQL/PhenotypeAdaptor.pm

```diff
@@ -236,6 +240,28 @@ sub _obj_from_row {
                                    mapping_type   => $row->{mapping_type} }) if defined $row->{accession} ;
 }
 
+=head2 fetch_phenotype_class_types
+
+  Example    : $phenotype_classes = $pheno_adaptor->fetch_phenotype_class_types();
+  Description: Retrieves the list of existing Phenotype class attribs
+               If no phenotype class type exists undef is returned.
+  Returntype : list ref of string
+  Exceptions : none
+  Caller     : general
+=cut
+
+sub fetch_phenotype_class_types {
+  my $self = shift;
+
+  my $phenos= $self->generic_fetch();
```

Comment:
Are you fetching all phenotypes here?

Reply:
Yes, currently I am. I was debating it I should do a distinct sql select, can change it to a sql.